### PR TITLE
Remove old aws-sdk gem

### LINKED ISSUE #9256: Json dependency of aws sdk gem is too old
## Description

`aws-sdk` is on version 1.67.0 on OFN. It requires `json ~> 1.4`. In the Gemfile OFN has `json` listed. When installing the project from scratch and following the getting started document, the OFN local instance ends up with json `2.6.1`.

When we run `bundle install`, it logs this message:

```bash
Your lockfile doesn't include a valid resolution.
You can fix this by regenerating your lockfile or trying to manually editing the bad locked gems to a version that satisfies all dependencies.
The unmet dependencies are:
* json (~> 1.4), depended upon aws-sdk-v1-1.67.0, unsatisfied by json-2.6.1
```

So, whenever we run `bundle install` (when trying to install a new gem, for example), we have the error:

```bash
Bundler could not find compatible versions for gem "json":
  In snapshot (Gemfile.lock):
    json (= 2.6.1)

  In Gemfile:
    json

    aws-sdk (= 1.67.0) was resolved to 1.67.0, which depends on
      aws-sdk-v1 (= 1.67.0) was resolved to 1.67.0, which depends on
        json (~> 1.4)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
``` 

## Expected Behavior

I would like to install new gems in the project

## Actual Behaviour

I cannot install any new gem

## Steps to Reproduce

1. Add a new gem in the Gemfile, say `phonelib` (in the context of #9249, for example )
2. Run `bundle install`

## Animated Gif/Screenshot

Screenshot from 2022-06-02 11-53-35

## Severity

bug-s2: a non-critical feature is broken, no workaround

## Your Environment

* Version used: master branch
* Browser name and version: -
* Operating System and version (desktop or mobile): Ubuntu 21.10

## Possible Fix

Update `aws-sdk` gem.

### PR DESCRIPTION
**Pull request by diegoamc**
#### What? Why?

Closes #9256 

The version used of `aws-sdk` had an old `json` dependency that prevented us to install new gems in the project. Updating the aws-sdk gem also updates the json gem, solving the issue.

#### What should we test?

- Test file upload on AWS
- Test file access like seeing an image previously uploaded on AWS

#### Release notes

Changelog Category: Remove old aws-sdk gem

### CODE REVIEW

**Comment by diegoamc** [comment]
I'm leaving this one as a draft because I don't really know how to test the AWS integration manually now. Does anyone have an easy way to test if it's still working? I don't think the automated tests would test this integration.

File: Gemfile.lock [hunk 1]
```diff
@@ -835,4 +829,4 @@ RUBY VERSION
    ruby 3.0.3p157
 
 BUNDLED WITH
-   2.1.4
+   2.2.32
```
Comment (mkllnk) [inline Gemfile.lock:832]: You are using a different bundler version here. Can you keep it the same for now? Upgrading bundler would be a different pull request.

**Review by mkllnk — APPROVED** [review]
Great! Thank you. Good timing as this was just unblocked. After this is merged, I would imagine that Dependabot will go crazy for a while. :wink: 

There's actually an automated test in the spree_image_spec. It doesn't test with a real server but mocks the network request/response which should test the aws-sdk, too.

For testing instructions:

- Deploy on a staging server with cloud storage.
- Check that previous images are still shown.
- Upload a new image.

**Comment by drummer83** [comment]
@filipefurtad0 I don't know how to test this. Can you have a go at this one?

**Comment by filipefurtad0** [comment]
Hey @drummer83 ,

Thanks for looking into this and reaching out. I'm happy to help.

Did you notice Maikel's additional steps on [how to test, on this comment](https://github.com/openfoodfoundation/openfoodnetwork/pull/9271#pullrequestreview-1005099801)?

It's useful to know, of course, that **staging-AU** uses AWS to store images in the cloud, so this would be the correct server for this test. You can also check this in the console, to see "who's the server talking to", when getting images:

image

I hope this unblocks you - please let us know this is not the case.

**Comment by sigmundpetersen** [comment]
Just a comment. Would backups be affected by this in any way?
Or is it not related?

**Comment by diegoamc** [comment]
> Just a comment. Would backups be affected by this in any way? Or is it not related?

@sigmundpetersen I think the aws-sdk update may affect backups as well. From what I could see, the project uses an updated `db2fog` fork. It doesn't seem to depend explicitly on `aws-sdk` but it uses AWS env vars. I think it's worth checking if backups still work in any case.

**Comment by mkllnk** [comment]
Thank you, @sigmundpetersen! We do need to check backups. uk-staging has backups configured, bucket name `ofn-uk-staging-backups-new`. Backups are run every 4 hours. So we can either wait for a while or run the command manually:
```
bundle exec rake 'db2fog:backup'
```

I think that we would see errors here if it wouldn't work. Ideally, we would check the bucket for a new backup file. But I can't see the AWS login details in Bitwarden.

Alternatively, I can just do a test in the development environment.

**Comment by mkllnk** [comment]
I'm not hopeful though. I just saw this comment in our Gemfile:

> gem "fog-aws", "~> 2.0" # db2fog does not support v3

Maybe we should migrate to the [backup](https://github.com/backup/backup) gem?

**Comment by mkllnk** [comment]
It worked! Somehow I had to add the mime-types gem to my dev environment, regardless of this pull request. But then the backup worked with and without the old aws-sdk-v1. :tada:

**Comment by mkllnk** [comment]
Okay, @drummer83, over to you again to test images.

**Comment by sigmundpetersen** [comment]
For reference, there is some previous work on adding the backup gem here #6837

**Comment by drummer83** [comment]
Thank you @diegoamc for this pull request and for keeping an eye on a clean installation of the Open Food Network! :muscle: 

I have tested this PR in the following way.

**Before staging this PR**
- images from aws are loading - here the ofn logo (the cat), the shop logo and the product image are loading correctly
grafik

**After staging this PR to our Australian staging server (which uses AWS for image storage)**
- the same images are still loading nicely
- as an enterprise user adding product images, enterprise logo and hero image works as expected
- as a super admin changing the OFN logos and footer logo works as expected
- as a customer viewing the added images works as expected
grafik

I think this one is ready to go! :partying_face: 
Thanks again!
